src/components/utility/dueDateSrv.js

```diff
@@ -6,18 +6,19 @@
         var WEEK = 7;
         this.SEVEN_DAYS_IN_MS = dayInMs * WEEK;
 
-        this.isDueDatePass = function (startDate) {
+
+        this.isDueDatePass = function (dueDate) {
```

Comment:
Look for all places we call "isDueDatePass"
Make sure we send the "dueDate" and not the "AssignDate"